File: game-service/src/main/java/io/github/xpakx/habitgame/battle/dto/BattleShip.java
package io.github.xpakx.habitgame.battle.dto;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class BattleShip {
    private Long id;
    private Long shipId;
    private String name;
    private String code;
    private Integer rarity;
    private Integer size;
    private Integer hp;
    private Integer strength;
    private Integer hitRate;
    private Integer criticalRate;
    private boolean destroyed;
    private boolean damaged;
    private boolean prepared;
    private boolean action;
    private boolean movement;
    private boolean enemy;
    private BattlePosition position;
}

File: frontend/src/app/battle/dto/battle-ship.ts
import { BattlePosition } from "./battle-position";

export interface BattleShip {
    id: number;
    shipId: number;
    name: String;
    code: String;
    rarity: number;
    size: number;
    hp: number;
    strength: number;
    hitRate: number;
    criticalRate: number;
    destroyed: boolean;
    damaged: boolean;
    prepared: boolean;
    action: boolean;
    movement: boolean;
    enemy: boolean;
    position: BattlePosition;
}